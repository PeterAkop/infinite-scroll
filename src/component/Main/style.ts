import {createStyles, makeStyles, Theme} from '@material-ui/core/styles';

export const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        paper: {
            padding: theme.spacing(3, 2),
            margin: 'auto',
            marginTop: '10%',
            width: '50%',
            display: 'flex',
            flexDirection: 'column'
        },
        title: {
            textAlign: 'center'
        },
        button: {
            margin: 'auto'
        }
    }),
);