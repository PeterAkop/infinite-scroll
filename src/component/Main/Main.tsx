import React, {ChangeEvent, FC, useState} from 'react';
import {Button, Paper, TextField, Typography} from '@material-ui/core';

import {useStyles} from './style'
import {history} from '../../App'


const Main: FC = () => {
        const [id, setId] = useState<string>('');

        const handleChange = (event: ChangeEvent<HTMLInputElement>) => setId(event.target.value);
        const handleClick = () => {
            history.push(`/contracts/${id}`);
        };

        const classes = useStyles();
        return (
            <Paper className={classes.paper}>
                <Typography
                    className={classes.title}
                    variant="h5"
                    component="h3"
                >
                    Enter contract id to view details
                </Typography>
                <TextField
                    id={'text-id'}
                    value={id}
                    onChange={handleChange}
                    label="Contract ID"
                    margin="normal"
                    variant="outlined"
                />
                <Button
                    disabled={id === ''}
                    className={classes.button}
                    variant="contained"
                    color="primary"
                    onClick={handleClick}
                >
                    View details
                </Button>
            </Paper>
        )
    }
;

export default Main;