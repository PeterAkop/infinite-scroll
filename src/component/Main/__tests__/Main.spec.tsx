import React from 'react';
import {shallow} from 'enzyme';

import Main from '../Main';


describe('Main', () => {

    it('should render correctly', () => {
        const component = shallow(<Main/>);

        expect(component).toMatchSnapshot();
    });

    it('text label should be Contract ID', () => {
        const component = shallow(<Main/>);

        expect(component.find('#text-id').props().label).toBe('Contract ID');
        expect(component).toMatchSnapshot();
    });


});
