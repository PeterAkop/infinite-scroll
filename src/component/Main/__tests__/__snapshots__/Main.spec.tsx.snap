// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render correctly 1`] = `ShallowWrapper {}`;

exports[`Main text label should be Contract ID 1`] = `ShallowWrapper {}`;
